### PR TITLE
Add doc for IAM permissions required by SkyPilot

### PR DESCRIPTION
**Pull request by infwinston**
Describe the changes in this PR:

To close https://github.com/skypilot-org/skypilot/issues/1443, we'd like to add some documents for required IAM permissions to run SkyPilot.

TODO in next PR:
- [ ] AWS IAM
- [ ] Azure IAM?

### CODE REVIEW

File: docs/source/getting-started/installation.rst [hunk 1]
```diff
@@ -75,6 +75,10 @@ If running :code:`conda install -c conda-forge google-cloud-sdk` produces the er
 
 Note: if you encounter *Authorization Error (Error 400: invalid_request)* with the url generated by :code:`gcloud auth login`, try installing the latest version of the `Google Cloud SDK <https://cloud.google.com/sdk/docs/install>`_ (e.g., with :code:`conda install -c conda-forge google-cloud-sdk`) on **your local machine** (which opened the browser) and rerun the command.
 
+.. note::
```
Comment (concretevitamin) [inline docs/source/getting-started/installation.rst:78]: This is already added?

File: docs/source/reference/tpu.rst [hunk 2]
```diff
@@ -86,6 +86,7 @@ Here is a complete task YAML that runs `MNIST training <https://cloud.google.com
          pip install "jax[tpu]>=0.2.16" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
          pip install --upgrade clu
          pip install -e flax
+         pip install tensorflow tensorflow-datasets
```
Comment (concretevitamin) [inline docs/source/reference/tpu.rst:89]: already added

File: docs/source/reference/faq.rst [hunk 3]
```diff
@@ -76,6 +76,29 @@ If you have edited the ``file_mounts`` section (e.g., by adding some files) and
...
+.. code-block:: yaml
+
+  roles/browser
+  roles/compute.admin
+  roles/iam.serviceAccountAdmin
+  roles/iam.serviceAccountUser
+  roles/serviceusage.serviceUsageConsumer
+  roles/storage.admin
+
+If you want to use TPUs, an additional role
```
Comment (concretevitamin) [inline docs/source/reference/faq.rst:93]:
```suggestion
Optionally, to use TPUs, an additional role is required:
```

File: docs/source/reference/faq.rst [hunk 4]
```diff
@@ -76,6 +76,29 @@ If you have edited the ``file_mounts`` section (e.g., by adding some files) and
...
+
+.. code-block:: yaml
+
+  roles/tpu.admin
+
+is required. You can grant those accesses via the `GCP IAM console <https://console.cloud.google.com/iam-admin/iam>`_.
```
Comment (concretevitamin) [inline docs/source/reference/faq.rst:99]:
```suggestion
You can grant those accesses via the `GCP IAM console <https://console.cloud.google.com/iam-admin/iam>`_.
```

File: docs/source/reference/faq.rst [hunk 5]
```diff
@@ -76,6 +76,29 @@ If you have edited the ``file_mounts`` section (e.g., by adding some files) and
 To avoid rerunning the ``setup`` commands, pass the ``--no-setup`` flag to ``sky launch``.
 
 
+What are the required IAM permissons on GCP for SkyPilot?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To use SkyPilot, your GCP account needs to be granted the following IAM roles.
```
Comment (concretevitamin) [inline docs/source/reference/faq.rst:82]:
```suggestion
To use SkyPilot, your GCP account needs to be granted the following IAM roles:
```

**Review by concretevitamin — COMMENTED** [review]
Thanks @infwinston. Some comments.

File: docs/source/reference/faq.rst (hunk 3 above)
Comment (ewzeng) [reply docs/source/reference/faq.rst:93]: I don't understand the suggested change. How can something be both optional and required?

Comment (concretevitamin) [reply docs/source/reference/faq.rst:93]: How about “Optionally, to use TPUs, add the following role”?

Comment (ewzeng) [reply docs/source/reference/faq.rst:93]: I think that is better. I still feel it is not clear that "optionally" is describing "to use TPUs", but maybe that's just me.

File: docs/source/getting-started/installation.rst (hunk 1 above)
Comment (infwinston) [reply docs/source/getting-started/installation.rst]: ah yes. just merged master!

**Comment by infwinston** [comment]
Sorry for the late response. I've updated the doc for the comments. Please let me know if it's good, thanks!

**Review by concretevitamin — APPROVED** [review]
Nice! Thanks @infwinston.

**Pull request merged**